### PR TITLE
Rework the fancy-hist.xsl, implement #875, fix #877 and some more improvementsFancy Hist Xsl Improvements

### LINKED ISSUE #875: Slight improvement request for fancy-hist.xsl
In the `History` tab of `fancy-hist.xsl`-generated report, I always only see the same information that is shown in the `Overall` line of the `Summary` tab.
I guess you can somehow combine reports on different versions and this will be shown in the `History` tab.
Unfortunately I didn't really find any info in the documentation on how to do that, maybe this could be improved too. :-)

The improvements I'd like to suggest right now are:
- Show the "Release" of the current analysis in the title line after the project name, not only in the history tab
- Fill empty cells in the "Summary" tab with `0` like in the "History" tab instead of leaving them empty
- Hide the History tab if there is no history but only the current analysis which just repeats the overall line of the summary tab
- In the two "Browse By ..." tabs, is a version dropdown, it is unclear what it is for, are there releases from the history selectable? If so, maybe either call it "Release" on all three, the commandline, the history tab and the dropdown or call it "Version" on all three, Calling it differently just confuses unnecessarily
- Hide the version selector dropdown in the two "Browse By ..." tabs if there is nothing to select actually
- Store the currently selected tab in the anchor of the URL, so that a reload of the page does not always jump back to the summary tab; this would also make the individual tabs bookmarkable

### LINKED ISSUE #877: BugInstance not properly handled by fancy-hist.xsl
I have seen the problem with this bug instance.
```xml
  <BugInstance type="OBL_UNSATISFIED_OBLIGATION" priority="2" rank="20" abbrev="OBL" category="EXPERIMENTAL" instanceHash="d37689192a435176894cf6ba76afd0e5" instanceOccurrenceNum="0" instanceOccurrenceMax="0">
    <ShortMessage>Method may fail to clean up stream or resource</ShortMessage>
    <LongMessage>new net.kautler.discord.logger.test.jre.ConfigLoader() may fail to clean up java.io.InputStream</LongMessage>
    <Class classname="net.kautler.discord.logger.test.jre.ConfigLoader" primary="true">
      <SourceLine classname="net.kautler.discord.logger.test.jre.ConfigLoader" start="39" end="66" sourcefile="ConfigLoader.java" sourcepath="net/kautler/discord/logger/test/jre/ConfigLoader.java" relSourcepath="java\net\kautler\discord\logger\test\jre\ConfigLoader.java">
        <Message>At ConfigLoader.java:[lines 39-66]</Message>
      </SourceLine>
      <Message>In class net.kautler.discord.logger.test.jre.ConfigLoader</Message>
    </Class>
    <Method classname="net.kautler.discord.logger.test.jre.ConfigLoader" name="&lt;init&gt;" signature="()V" isStatic="false" primary="true">
      <SourceLine classname="net.kautler.discord.logger.test.jre.ConfigLoader" start="46" end="66" startBytecode="0" endBytecode="466" sourcefile="ConfigLoader.java" sourcepath="net/kautler/discord/logger/test/jre/ConfigLoader.java" relSourcepath="java\net\kautler\discord\logger\test\jre\ConfigLoader.java"/>
      <Message>In method new net.kautler.discord.logger.test.jre.ConfigLoader()</Message>
    </Method>
    <Class classname="java.io.InputStream" role="CLASS_REFTYPE">
      <SourceLine classname="java.io.InputStream" start="50" end="708" sourcefile="InputStream.java" sourcepath="java/io/InputStream.java">
        <Message>At InputStream.java:[lines 50-708]</Message>
      </SourceLine>
      <Message>Reference type java.io.InputStream</Message>
    </Class>
    <Int value="1" role="INT_OBLIGATIONS_REMAINING">
      <Message>1 instances of obligation remaining</Message>
    </Int>
    <SourceLine classname="net.kautler.discord.logger.test.jre.ConfigLoader" primary="true" start="48" end="48" startBytecode="19" endBytecode="19" sourcefile="ConfigLoader.java" sourcepath="net/kautler/discord/logger/test/jre/ConfigLoader.java" relSourcepath="java\net\kautler\discord\logger\test\jre\ConfigLoader.java" role="SOURCE_LINE_OBLIGATION_CREATED">
      <Message>Obligation to clean up resource created at ConfigLoader.java:[line 48] is not discharged</Message>
    </SourceLine>
    <SourceLine classname="net.kautler.discord.logger.test.jre.ConfigLoader" start="49" end="49" startBytecode="25" endBytecode="25" sourcefile="ConfigLoader.java" sourcepath="net/kautler/discord/logger/test/jre/ConfigLoader.java" relSourcepath="java\net\kautler\discord\logger\test\jre\ConfigLoader.java" role="SOURCE_LINE_PATH_CONTINUES">
      <Message>Path continues at ConfigLoader.java:[line 49]</Message>
    </SourceLine>
    <SourceLine classname="net.kautler.discord.logger.test.jre.ConfigLoader" start="63" end="63" startBytecode="155" endBytecode="155" sourcefile="ConfigLoader.java" sourcepath="net/kautler/discord/logger/test/jre/ConfigLoader.java" relSourcepath="java\net\kautler\discord\logger\test\jre\ConfigLoader.java" role="SOURCE_LINE_PATH_CONTINUES">
      <Message>Path continues at ConfigLoader.java:[line 63]</Message>
    </SourceLine>
    <SourceLine classname="net.kautler.discord.logger.test.jre.ConfigLoader" start="64" end="64" startBytecode="163" endBytecode="163" sourcefile="ConfigLoader.java" sourcepath="net/kautler/discord/logger/test/jre/ConfigLoader.java" relSourcepath="java\net\kautler\discord\logger\test\jre\ConfigLoader.java" role="SOURCE_LINE_PATH_CONTINUES">
      <Message>Path continues at ConfigLoader.java:[line 64]</Message>
    </SourceLine>
    <SourceLine classname="net.kautler.discord.logger.test.jre.ConfigLoader" start="65" end="65" startBytecode="169" endBytecode="169" sourcefile="ConfigLoader.java" sourcepath="net/kautler/discord/logger/test/jre/ConfigLoader.java" relSourcepath="java\net\kautler\discord\logger\test\jre\ConfigLoader.java" role="SOURCE_LINE_PATH_CONTINUES">
      <Message>Path continues at ConfigLoader.java:[line 65]</Message>
    </SourceLine>
    <SourceLine classname="net.kautler.discord.logger.test.jre.ConfigLoader" start="66" end="66" startBytecode="186" endBytecode="186" sourcefile="ConfigLoader.java" sourcepath="net/kautler/discord/logger/test/jre/ConfigLoader.java" relSourcepath="java\net\kautler\discord\logger\test\jre\ConfigLoader.java" role="SOURCE_LINE_PATH_CONTINUES">
      <Message>Path continues at ConfigLoader.java:[line 66]</Message>
    </SourceLine>
    <String value="{InputStream x 1}" role="STRING_REMAINING_OBLIGATIONS">
      <Message>Remaining obligations: {InputStream x 1}</Message>
    </String>
  </BugInstance>
```

- it is included in the "Summary" tab properly
- it is included in the "History" tab properly
- it is included in the "Browse By Categories" tab properly
- it is **missing** in the "Browse By Packages" tab

### PR DESCRIPTION
**Pull request by Vampire**
Closes #875
Closes #877

### CODE REVIEW

**Comment by KengoTODA** [comment]
@spotbugs/everyone I'm not good at XSL, is there any other member who can review this PR?

File: spotbugs/src/xsl/fancy-hist.xsl [hunk 1]
```diff
@@ -246,20 +282,18 @@
...
-            var versionsBox = document.findbugsForm.versions.options;
-            versionsBox[0] = new Option(" -- All Versions -- ","-1");
-            versionsBox.selectedIndex = 0;
-            if (versions.length>=1) {
-               for (x=0; versions.length>1 && x<versions.length; x++) {
-                  versionsBox[x+1] = new Option(" Bugs at release: "+versions[versions.length-x-1][1], versions[versions.length-x-1][0]);
-               }
+            var versionsBox = document.findbugsForm.versions;
+            versionsBox.options[0] = new Option(" -- All Releases -- ", "-1");
+            versionsBox.options[0].id = "-1";
```
[the remark this answers is not in the document]
Comment (Vampire) [reply spotbugs/src/xsl/fancy-hist.xsl]: Well, the value was like that before, I just changed that the id field is also filled, for easier persistence of the state in the URL anchor.
Actually I think most people are not aware that you can have multiple versions in one report anyway and I also didn't find documentation on how to achieve this, so the versionsBox will be hidden in most cases anyway.

**Comment by Vampire** [comment]
> please update `CHANGELOG.md` too.

done

**Comment by Vampire** [comment]
> Please try to avoid force-pushes when possible.

May I ask why?
force-pushes are quite common when incorporating PR-feedback
I usually prefer having few complete commits instead of many correction commits

**Comment by henrik242** [comment]
Force push *in a PR* is perfectly acceptable, even preferable in some cases, IMHO. I don't understand why it should be avoided.

**Pull request merged**